packages/@aws-cdk/aws-synthetics/test/canary.test.ts

```diff
@@ -0,0 +1,223 @@
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    ExecutionRoleArn: stack.resolve(role.roleArn),
+  });
+});
+
+test('An existing bucket can be specified instead of auto-created', () => {
+  // GIVEN
+  const bucket = new s3.Bucket(stack, 'mytestbucket');
+  const prefix = 'canary';
+
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    artifactsBucketLocation: { bucket, prefix },
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    ArtifactS3Location: stack.resolve(bucket.s3UrlForObject(prefix)),
+  });
+});
+
+test('Runtime can be specified', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    runtime: synthetics.Runtime.SYNTHETICS_1_0,
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    RuntimeVersion: 'syn-1.0',
+  });
+});
+
+test('Runtime can be customized', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    runtime: new synthetics.Runtime('syn-1.0'),
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    RuntimeVersion: 'syn-1.0',
+  });
+});
+
+test('Schedule can be set with Rate', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    schedule: synthetics.Schedule.rate(Duration.minutes(3)),
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Schedule: objectLike({ Expression: 'rate(3 minutes)'}),
+  });
+});
+
+test('Schedule can be set with Expression', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    schedule: synthetics.Schedule.expression('rate(1 hour)'),
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Schedule: objectLike({ Expression: 'rate(1 hour)'}),
+  });
+});
+
+test('Schedule can be set to run once', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    schedule: synthetics.Schedule.once(),
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Schedule: objectLike({ Expression: 'rate(0 minutes)'}),
+  });
+});
+
+test('Throws when rate above 60 minutes', () => {
+  expect(() => new synthetics.Canary(stack, 'Canary', {
```

Comment:
I think it might be useful to have a test that shows what happens when the duration isn't whole minutes... Like `Duration.seconds(59)`.